src/observability_utils/tracing/helpers.py

```diff
@@ -41,11 +43,20 @@ def setup_tracing(name: str, with_otlp_export: bool = True) -> None:
         name (str): The name to be used in spans to refer to the application.
         with_otlp_export (bool): Indicates whether an OTLP Exporter shoudld be set up
     """
-    resource = Resource(attributes={"service.name": name})
-    provider = TracerProvider(resource=resource)
-    if with_otlp_export:
-        provider.add_span_processor(BatchSpanProcessor(OTLPSpanExporter()))
-    set_tracer_provider(provider)
+
+    # Only create a TracerProvider if one does not already exist, in which case
+    # get_trace_provider will return a ProxyTracerProvider
+    if isinstance(get_tracer_provider(), ProxyTracerProvider):
+        resource = Resource(
+            attributes={
+                "service.name": name,
+                "service.instrument": environ.get("BEAMLINE", "Unknown"),
+            }
+        )
```

Comment:
```suggestion
        instrument = environ.get("INSTRUMENT", environ.get("BEAMLINE", "Unknown"))
        resource = Resource(
            attributes={
                "service.name": name,
                "service.instrument": instrument,
                "service.beamline": instrument,
            }
        )
```

Do we want to support the old term `beamline` *in the tracing where it hasn't been before?*

Do we want to start to support INSTRUMENT as an environment variable to try and eventually transition away from BEAMLINE? I recognise both suggestions are at odds with one another, I'm not certain which way we should go. 

blueapi does the following:

```python
    instrument: str = os.environ.get("BEAMLINE", os.environ.get("INSTRUMENT", "dev"))

    def filter(self, record: logging.LogRecord) -> bool:
        record.instrument = self.instrument
        record.beamline = self.instrument
        return True
```

Reply:
Ill adjust to make sconsistent with blueapi